### PR TITLE
[Docs] Add docstring for AlertConfig

### PR DESCRIPTION
**Pull request by Yacouby**
resolves:
https://iguazio.atlassian.net/browse/ML-7252

### CODE REVIEW

File: mlrun/alerts/alert.py [hunk 1]
```diff
@@ -54,6 +54,62 @@ def __init__(
...
+        :param criteria:       how many times this alert was triggered and in what time period
+        :param reset_policy:   when to clear the alert. May be manual for manual reset of the alert, or
+                               auto if the criteria contains a time period
+        :param notifications:  list of notifications to invoke once the alert is triggered
+        :param entities:       entities that the event relates to. The entity object will contain fields that uniquely
+                               identify a given entity in the system
+        :param id:             internal id of the alert (user should not supply it)
+        :param state:          state of the alert, may be active/inactive (user should not supply it)
+        :param created:        when the alert is created (user should not supply it)
+        :param count:          internal counter of the alert (user should not supply it)
```
Comment (TomerShor) [inline mlrun/alerts/alert.py:118]: If a user should not supply it, why do we expose it in the init?

File: mlrun/alerts/alert.py [hunk 2]
```diff
@@ -54,6 +54,62 @@ def __init__(
...
+            )
+            project.store_alert_config(alert_data)
+
+        :param project:        name of the project to associate the alert with
+        :param name:           name of the alert
+        :param description:    description of the alert
+        :param summary:        summary of the alert, will be sent in the generated notifications
+        :param severity:       severity of the alert
+        :param trigger:        the events that will trigger this alert, may be a simple trigger based on events or
+                               complex which is based on a prometheus alert
```
Comment (TomerShor) [inline mlrun/alerts/alert.py]:
```suggestion
                               complex trigger which is based on a prometheus alert
```

File: mlrun/alerts/alert.py [hunk 3]
```diff
@@ -54,6 +54,62 @@ def __init__(
...
+        :param criteria:       how many times this alert was triggered and in what time period
```
Comment (TomerShor) [inline mlrun/alerts/alert.py]: How many time it **was** triggered? Or **will be** triggered?

File: mlrun/alerts/alert.py [hunk 4]
```diff
@@ -54,6 +54,62 @@ def __init__(
...
+        :param reset_policy:   when to clear the alert. May be manual for manual reset of the alert, or
+                               auto if the criteria contains a time period
```
Comment (TomerShor) [inline mlrun/alerts/alert.py]:
```suggestion
        :param reset_policy:   when to clear the alert. May be "manual" for manual reset of the alert, or
                               "auto" if the criteria contains a time period
```

File: mlrun/alerts/alert.py [hunk 5]
```diff
@@ -54,6 +54,62 @@ def __init__(
         created: str = None,
         count: int = None,
     ):
+        """
+        Alert config object
+
+        Example::
+            # create an alert on endpoint_id, which will be triggered to slack if there was an event of:
+            "data_drift_detected" 3 times in the next hour.
```
Comment (TomerShor) [inline mlrun/alerts/alert.py]:
```suggestion
            # create an alert on endpoint_id, which will be triggered to slack if there is a
            # "data_drift_detected" event 3 times in the next hour.
```

File: mlrun/alerts/alert.py [hunk 6]
```diff
@@ -54,6 +54,62 @@ def __init__(
...
+            from mlrun.alerts.alert import AlertConfig
```
Comment (TomerShor) [inline mlrun/alerts/alert.py]: General question - Can we also import from the parent module?
Meaning `mlrun.alerts.AlertConfig`?

File: mlrun/alerts/alert.py [hunk 7]
```diff
@@ -54,6 +54,62 @@ def __init__(
...
+            kind="slack",
+            name="slack_notification",
+            message="drift was detected",
+            severity="warning",
+            when=["now"],
+            condition="failed",
+            secret_params={
+                "webhook": "https://hooks.slack.com/",
+            },
+            ).to_dict()
```
Comment (TomerShor) [inline mlrun/alerts/alert.py]: Fix indentation:
```suggestion
            notification = mlrun.model.Notification(
                kind="slack",
                name="slack_notification",
                message="drift was detected",
                severity="warning",
                when=["now"],
                condition="failed",
                secret_params={
                    "webhook": "https://hooks.slack.com/",
                },
            ).to_dict()
```

File: mlrun/alerts/alert.py [hunk 8]
```diff
@@ -54,6 +54,62 @@ def __init__(
...
+            name="drift-alert",
+            summary="a drift was detected",
+            severity=alert_objects.AlertSeverity.LOW,
+            entities=alert_objects.EventEntities(
+                kind=entity_kind, project="my-project", ids=[entity_id]
+            ),
+            trigger=alert_objects.AlertTrigger(events=[event_name]),
+            criteria=alert_objects.AlertCriteria(count=3, period="1h"),
+            notifications=[alert_objects.AlertNotification(notification=notification)],
+            )
```
Comment (TomerShor) [inline mlrun/alerts/alert.py]: Same here:
```suggestion
            alert_data = AlertConfig(
                project="my-project",
                name="drift-alert",
                summary="a drift was detected",
                severity=alert_objects.AlertSeverity.LOW,
                entities=alert_objects.EventEntities(
                    kind=entity_kind, project="my-project", ids=[entity_id]
                ),
                trigger=alert_objects.AlertTrigger(events=[event_name]),
                criteria=alert_objects.AlertCriteria(count=3, period="1h"),
                notifications=[alert_objects.AlertNotification(notification=notification)],
            )
```

**Review by TomerShor — CHANGES_REQUESTED** [review]
Overall looking good, left some suggestions for improvement

File: mlrun/alerts/alert.py (hunk 6 above)
Comment (Yacouby) [reply mlrun/alerts/alert.py]: yes

File: mlrun/alerts/alert.py (hunk 1 above)
Comment (Yacouby) [reply mlrun/alerts/alert.py:118]: these fields are returned in the get alert operation, so they need to be initialized in the client side too

File: mlrun/alerts/alert.py (hunk 6 above)
Comment (TomerShor) [reply mlrun/alerts/alert.py]: So let's change it to
```suggestion
            from mlrun.alerts import AlertConfig
```

File: mlrun/alerts/alert.py (hunk 1 above)
Comment (TomerShor) [reply mlrun/alerts/alert.py:118]: You mean in `from_dict`?

**Review by TomerShor — CHANGES_REQUESTED** [review]

File: mlrun/alerts/alert.py (hunk 1 above)
Comment (Yacouby) [reply mlrun/alerts/alert.py:118]: yes

File: server/api/db/sqldb/db.py [hunk 9]
```diff
@@ -5115,6 +5115,7 @@ def enrich_alert(self, session, alert: AlertConfig):
             else mlrun.common.schemas.AlertActiveState.INACTIVE
         )
         alert.count = state.count
+        alert.created = datetime.now(timezone.utc)
```
Comment (TomerShor) [inline server/api/db/sqldb/db.py]: `enrich_alert` is also called in `get_enriched_alert`, and there you don't want to override the created field.
move it to `_create_alert`

**Review by TomerShor — CHANGES_REQUESTED** [review]
BTW this change for the `created` field can be in another PR, and we'll leave this PR only for Docs change

**Review by TomerShor — APPROVED** [review]

**Pull request merged**